test/rb/generate_data.py

```diff
@@ -0,0 +1,598 @@
+                                                     shots=shots,
+                                                     noise_model=noise_model,
+                                                     seed_simulator=SEED).result())
+    cnotdihedral_z_results = []
+    for circuit in rb_cnotdihedral_z_circs:
+        cnotdihedral_z_results.append(qiskit.execute(circuit, backend=backend,
+                                                     basis_gates=basis_gates,
+                                                     shots=shots, noise_model=noise_model,
+                                                     seed_simulator=SEED).result())
+    return cnotdihedral_x_results, xdata, cnotdihedral_z_results
+
+
+def rb_purity_circuit_execution(rb_opts: dict, shots: int):
+    """
+        Create purity rb circuits with depolarizing errors and simulates them
+
+        Args:
+            rb_opts: the options for the rb circuits
+            shots: number of shots for each circuit simulation
+
+        Returns:
+            list: list of Results of the purity circuits simulations
+            list: the xdata of the rb circuit
+            int: npurity (3^(number of qubits))
+            list: list of Results of the coherent circuits simulations
+
+        """
+    # Load simulator
+    backend = qiskit.Aer.get_backend('qasm_simulator')
+    basis_gates = ['u1', 'u2', 'u3', 'cx']
+
+    rb_purity_circs, xdata, npurity = rb.randomized_benchmarking_seq(**rb_opts)
+
+    # Add depolarizing noise to the simulation
+    noise_model = NoiseModel()
+    p1q = 0.002
+    p2q = 0.01
+    noise_model.add_all_qubit_quantum_error(depolarizing_error(p1q, 1), 'u2')
+    noise_model.add_all_qubit_quantum_error(depolarizing_error(2 * p1q, 1), 'u3')
+    noise_model.add_all_qubit_quantum_error(depolarizing_error(p2q, 2), 'cx')
+
+    # coherent noise
+    err_unitary = np.zeros([2, 2], dtype=complex)
+    angle_err = 0.1
+    for i in range(2):
+        err_unitary[i, i] = np.cos(angle_err)
+        err_unitary[i, (i + 1) % 2] = np.sin(angle_err)
+    err_unitary[0, 1] *= -1.0
+
+    error = coherent_unitary_error(err_unitary)
+    coherent_noise_model = NoiseModel()
+    coherent_noise_model.add_all_qubit_quantum_error(error, 'u3')
+
+    purity_results = []
+    coherent_results = []
+    for circuit_list in rb_purity_circs:
+        for purity_num in range(npurity):
+            current_circ = circuit_list[purity_num]
+            # non-coherent purity results
+            purity_results.append(qiskit.execute(current_circ, backend=backend,
+                                                 basis_gates=basis_gates,
+                                                 shots=shots,
+                                                 noise_model=noise_model,
+                                                 seed_simulator=SEED).result())
+            # coherent purity results
+            coherent_results.append(qiskit.execute(current_circ, backend=backend,
```

Comment:
I don't think that we actually check the coherent error fitter, since it's too noisy, but maybe it's good to keep it as a reference.